txfixtures/tachandler.py

```diff
@@ -105,19 +105,23 @@ def setUp(self, spew=False, umask=None,
             stdout=subprocess.PIPE,
             stderr=subprocess.STDOUT,
             )
-        self.addCleanup(self.killTac)
-        stdout = until_no_eintr(10, self._proc.stdout.read)
-        if stdout:
-            raise TacException('Error running %s: unclean stdout/err: %s'
-                               % (args, stdout))
-        rv = self._proc.wait()
-        # twistd will normally fork off into the background with the
-        # originally-spawned process exiting 0.
-        if rv != 0:
-            raise TacException('Error %d running %s' % (rv, args))
-
-        self.addDetail(self.logfile, content_from_file(self.logfile))
-        self._waitForDaemonStartup()
+        try:
+            self.addCleanup(self.killTac)
```

Comment:
Minor nit, but I believe the call to addCleanup could be left out of the try/except block, as it was originally.

Reply:
Done.